main.cpp

```diff
@@ -183,105 +298,263 @@ void Vinyl::moveToneArm(bool is12Inch)
-    bool subOscOn = false;
-    int octave = 0; 
+    Oscillator();
+    double sawLevel;
+    double squLevel;
+    double tuning;
+    double subOscOn;
+    int octave;
 
     double osc1FM(double osc1SawSignal, double osc1SquSignal);
-    double subOscSignal(bool subOscOn);
+    double subOscSignal(bool isSubOscOn);
     int octaveLED(int octave, bool shiftButtonPressed);
 };
 
+Oscillator::Oscillator()
+{
+    sawLevel = 0.0;
+    squLevel = 1.0;
+    tuning = 0.0;
+    subOscOn = 0.0;
+    octave = 0; 
+    std::cout << "Instance of 'Oscillator' has been constructed" << std::endl;
+}
+
 double Oscillator::osc1FM(double osc1SawSignal, double osc1SquSignal)
 {
     return osc1SawSignal * osc1SquSignal;
 }
 
+double Oscillator::subOscSignal(bool isSubOscOn)
+{ 
+    double subsignal = 0.5;
+    subOscOn = isSubOscOn ? 1.0 : 0.0;
+    return subsignal * subOscOn;
+}
+
 int Oscillator::octaveLED(int selectedOctave, bool shiftButtonPressed)
 {
-    selectedOctave = shiftButtonPressed ? selectedOctave : selectedOctave - 4;
+    selectedOctave = shiftButtonPressed ? selectedOctave -1 : selectedOctave;
     return selectedOctave;
 }
 
 struct LFO
 {
-    double freqLFO = 30;
-    bool squLFO = false;
-    double attenuate = 1.0; 
-    bool speedSlow = true;
-    int routingLFO = 0;
+    LFO();
+    double freqLFO, attenuate;
+    bool squLFO, speedSlow;
+    int routingLFO;
 
     double modCutoff(double attenuate, double LFOSignal);
-    double modPitch(int routingLFO, double LFOSignal);
-    double modSAH(int routingLFO, double LFOSignal);
+    double modPitch(int LFOdestination, double LFOSignal);
+    double modSAH(int LFOdestination, double LFOSignal);
 };
 
+LFO::LFO()
+{
+    freqLFO = 30;
+    squLFO = false;
+    attenuate = 1.0; 
+    speedSlow = true;
+    routingLFO = 0;
+    std::cout << "Instance of 'LFO' has been constructed" << std::endl;
+}
+
 double LFO::modCutoff(double attenuateLevel, double LFOSignal)
 {
-    return LFOSignal * attenuateLevel;
+    return LFOSignal - attenuateLevel;
 }
 
+double LFO::modPitch(int LFOdestination, double LFOSignal)
+{
+    if (LFOdestination == 0)
+    {
+        return LFOSignal;
+    }
+    return 0;
+}
 
+double LFO::modSAH(int LFOdestination, double LFOSignal)
+{
+    if (LFOdestination == 1)
+    {
+        return LFOSignal;
+    }
+    return 0;
+}
 
 struct Envelope
 {
-    int attack = 5;
-    int decay = 100;
-    double sustain = 0.5;
-    int release = 500;
-
-    double oscGainEnvelope(double envelopeSignal, int routingEnvelope = 0 );
-    double cutoffEnvelope(double envelopeSignal, int routingEnvelope = 1 );
-    int routingControl(double envelopeSignal, int routingEnvelope = 2 );
+    Envelope(int initAttack, int initDecay, double initSustain, int initRelease);
+    int attack;
+    int decay;
+    double sustain;
+    int release;
+    double output;
+
+    double oscGainEnvelope(double amtToVCA);
+    void loopEnvelope(bool loop);
+    void invertEnvelope(bool invert);
```

Comment:
These function names do not describe an action or verb.  
You wrote a noun for the function name

- [ ] revise